components/ai_chat/core/browser/conversation_handler_unittest.cc

```diff
@@ -1183,6 +1183,67 @@ TEST_F(ConversationHandlerUnitTest_NoAssociatedContent, GenerateQuestions) {
   testing::Mock::VerifyAndClearExpectations(engine);
 }
 
+TEST_F(ConversationHandlerUnitTest, SelectedLanguage) {
+  MockEngineConsumer* engine = static_cast<MockEngineConsumer*>(
+      conversation_handler_->GetEngineForTesting());
+
+  std::string expected_input1 = "Now stand aside, worthy adversary.";
+  std::string expected_input2 = "A scratch? Your arm's off!";
+  std::string expected_selected_language = "fr";
+
+  testing::Sequence s;
+  EXPECT_CALL(*engine,
+              GenerateAssistantResponse(false, StrEq(""), _, expected_input1,
+                                        StrEq(""), _, _))
+      .Times(1)
+      .WillOnce(::testing::DoAll(
+          base::test::RunOnceCallback<5>(
+              mojom::ConversationEntryEvent::NewCompletionEvent(
+                  mojom::CompletionEvent::New("Tis but a scratch."))),
+          base::test::RunOnceCallback<5>(
+              mojom::ConversationEntryEvent::NewSelectedLanguageEvent(
+                  mojom::SelectedLanguageEvent::New(
+                      expected_selected_language))),
+          base::test::RunOnceCallback<6>(base::ok(""))));
+  EXPECT_CALL(*engine, GenerateAssistantResponse(
+                           false, StrEq(""), _, expected_input2,
+                           StrEq(expected_selected_language), _, _))
+      .WillOnce(::testing::DoAll(
+          base::test::RunOnceCallback<5>(
+              mojom::ConversationEntryEvent::NewCompletionEvent(
+                  mojom::CompletionEvent::New("No, it isn't."))),
+          base::test::RunOnceCallback<6>(base::ok(""))));
+
+  conversation_handler_->PerformAssistantGeneration(expected_input1, "", false,
+                                                    "");
+  conversation_handler_->PerformAssistantGeneration(expected_input2, "", false,
+                                                    "");
+
+  // Selected Language events should not be added to the conversation events
+  // history
+  const auto& conversation_history =
+      conversation_handler_->GetConversationHistory();
+  ASSERT_FALSE(conversation_history.empty());
+  bool has_selected_language_event = std::any_of(
+      conversation_history.begin(), conversation_history.end(),
+      [](const auto& conversation) {
+        return std::any_of(conversation->events->begin(),
```

Comment:
We normally use `base::ranges::any_of`

Reply:
Fixed in a63bdca4187730e71a9d70be8a33135c1951795d